pipeline/tests/asm.rs

```diff
@@ -263,9 +270,6 @@ fn dynamic_vadcop() {
     assert_eq!(witness_by_name["main_arith::y"].len(), 32);
     assert_eq!(witness_by_name["main_memory::m_addr"].len(), 32);
 
-    // Because machines have different lengths, this can only be proven
-    // with a composite proof.
-    run_pilcom_with_backend_variant(pipeline_gl.clone(), BackendVariant::Composite).unwrap();
```

Comment:
Not sure if it is important to test PILCOM here? If we already have the mock prover? If yes, we'd have to make a new pipeline with `LinkerMode::Native` (which implies re-computing everything until the witness).

With a bus, PILCOM ends in some infinite loop. See [this CI run](https://github.com/powdr-labs/powdr/actions/runs/12547802014/job/34986476973?pr=2289).